%!TEX root = ../thesis.tex
%*******************************************************************************
%*********************************** First Chapter *****************************
%*******************************************************************************

\chapter{Cellular identity in the genomics era}  %Title of the First Chapter

\ifpdf
    \graphicspath{{Chapter1/Figs/Raster/}{Chapter1/Figs/PDF/}{Chapter1/Figs/}}
\else
    \graphicspath{{Chapter1/Figs/Vector/}{Chapter1/Figs/}}
\fi

Cell biologists have attempted, from the inception of the discipline, to categorize the extensive variability of cells that are found in Nature. This endeavour is hampered by the intrinsic complexity of cells, which associated to their small size and sensitivity to the surrounding environment, makes cellular phenotypes hard to probe in an integrated and comprehensive way. The last decade however has seen extraordinary improvements in the detail to which molecules are assayed in individual cells. Single-cell sequencing, and in particular single-cell RNA-seq (scRNA-seq), has for the first time provided an unbiased, transcriptome-wide census of RNA molecules, allowing cells to be grouped by gene expression, which underlies most biological processes that influence cell function and identity. Extracting these groups and from the massive amounts of transcriptome sequencing data produced has in parallel required the continuous adoption of new computational and analytical methodologies.

This chapter provides a historical introduction to the definition of cell types, and how more recently developed experimental and computational approaches are shaping our understanding of how cells are categorized.

\nomenclature[z-scRNA-seq]{scRNA-seq}{Single-cell RNA sequencing}
\nomenclature[z-RNA]{RNA}{Ribonucleic acid}

%*************************** %First Section  *******************************
\section{Cell type discovery and definition} %Section - 1.1 
\label{section1.3}

%% this paragraph introduces cell biology
The term "cell" was coined by Robert Hooke in the 17th century to describe the empty cell walls he observed in cork samples through his microscope~\citep{hooke_micrographia:_1667}. This observation was complemented some years later, when Antonie van Leeuwenhoek first observed live unicellular organisms and cells with a microscope compose of more powerful lenses~\citep{mazzarello_unifying_1999}. Research and observations in the following 200 years led to the formulation of cell theory. Its first tenet was introduced by Schleiden and Schwann, and states that that all living structures are composed of cells or their byproducts~\citep{schwann_microscopical_1847}. The theory was later complemented by Robert Remak, Rudolf Virchow, and Albert KÃ¶lliker to include the postulate that all cells are derived from other cells (in the latin formulation popularized by Virchow, \textit{omnis cellula e cellula}).

%% this paragraph introduces how cellular diversity was classically studied with microscopy
These early studies looked at a variety of sources to unveil different types of cells. Leeuwenhoek reported observations from blood, brain, muscle and semen~\citep{leeuwenhoeck_m_microscopical_1674,leeuwenhoek_antoni_van_observationes_1677}. Subsequent developments of microscopy techniques led to improved imaging of a variety of tissues and the cells that compose them. For the first centuries of cell biology, microscopy was the method of choice to identify cell types. While this was mostly due to the relatively reduced knowledge of cellular biochemistry, it was immediately apparent that cellular morphology was intrinsically tied to its function. The most illustrative example of this is the neuron, whose unique structure was only unravelled after subsequent improvements in tissue preparation and staining, as well as increases in resolution and development of electron microscopy~\citep{mazzarello_unifying_1999}. These advances were also crucial to the identification of organelles. While larger structures, like nuclei, are still identifiable with simpler microscopes~\citep{brown_organs_1866}, others required improved resolution and staining or preparation to be identified~\citep{golgi_structure_1989}. Other advancements in microscopy like live-cell imaging or super resolution microscopy are constantly perfected to expand the boundaries of cellular functional characterization.

%% this paragraph introduces how molecular phenotyping started being used to define cell identity
Developments in biochemistry and molecular biology revealed that most organic molecules that compose cells are directly responsible for their function. Proteins are responsible for most cellular functions, being involved in enzymatic reactions, signalling and regulatory pathways or structural components. They became a prime target for cellular phenotyping with the development of immunostaining~\citep{coons_immunological_1941}, whereby an antibody that specifically targets a certain protein is usually tagged with a fluorophore. Immunostaining can identify protein expression in tissue slices, and the use of different fluorophores allows for the imaging of cells expressing multiple proteins. The usefulness of immunostaining became especially apparent when it was combined with high-throughput microfluidics methods and used for fluorescence-activated cell sorting (FACS)~\citep{bonner_fluorescence_1972}. This introduced the first high-throughput studies on molecular phenotyping of cells, and sorting allowed cell function to be probed in parallel~\citep{julius_demonstration_1972}. More recently, mass cytometry has allowed for a further expansion of the repertoire of proteins assayed~\citep{bandura_mass_2009,di_palma_unraveling_2015}. This technique, while destructive, has also been combined with imaging, adding a spatial component to the cell populations examined.

%% this last paragraph lists the gaps that had to be filled by genomics for cell biology
The identification and classification of cell types is dependent on their function. While function is deeply related to cellular morphology~\citep{prasad_cell_2019}, it is only a consequence of the pathways molecular pathways shaping it. Additionally, even though recent advances permit high throughput cell sorting through imaging~\citep{nitta_intelligent_2018}, the limited resolution hinders the identification of finer details of cell and organelle shape, which are frequently more informative of cellular activity. Cell sorting with fluorescent antibodies and mass cytometry can reveal more details on the molecules underlying cellular behaviour, but they are targeted approaches that depend on prior knowledge of the effector molecules. The more recent attempts at defining cell identity have therefore relied on the unbiased, high-throughput character of single-cell RNA-sequencing methods.


\nomenclature[z-FACS]{FACS}{Fluorescence-Activated Cell Sorting}

%********************************** %Second Section  *************************************
\section{Defining cell types using scRNA-seq} %Section - 1.2
\label{section1.2}

%% history of the growth of scRNA-seq
Methods to sequence the transcriptome of individual cells started to be developed shortly after the advent of RNA-seq~\citep{mortazavi_mapping_2008,tang_mrna-seq_2009}. This early development was pushed not by a need to define the molecular makeup of the unit of life, but rather to allow transcriptomic studies to be performed in low-input samples. Nonetheless, this seminal work still sparked the improvements that occurred in the decade that followed~\citep{svensson_exponential_2018}.

Initial developments focused on increasing sensitivity, since the original protocol was performed on cells from very early developmental stages, when they are larger and contain more RNA than most differentiated cell types. Different methodologies quantified gene expression by sequencing distinct transcript segments (either the 5' or the 3' end, or the full transctipt)~\citep{islam_characterization_2011,hashimshony_cel-seq:_2012,ramskold_full-length_2012,picelli_full-length_2014}. The idea of multiplexed scRNA-seq also started gaining traction (see Section \ref{section1.3}) with the use of multi-well plates or molecular barcodes for cells. The company Fluidigm eventually introduced the first commercially available microfluidics chips (called the "Fluidigm C1 system") for miniaturized cell isolation, RNA extraction and reverse transcription~\citep{brennecke_accounting_2013}. It is from this point that increased cell capture becomes the major technological driver. The major contributors to this have been nanodroplet-based technologies, that have put the number of profiled cells per dataset in the range of 10.000 to 100.000~\citep{macosko_highly_2015,klein_droplet_2015}. The importance of this increase on throughput has been demonstrated by Shekar and colleagues~\citep{shekhar_comprehensive_2016}, where they demonstrate that a Drop-seq dataset of approximately 25.000 cells sequenced at low depth could identify more \textit{bona fide} clusters than a smaller, more deeply sequenced Smart-seq2 dataset. Currently, most single-cell RNA-seq datasets use droplet-based technologies, chiefly the protocols designed for the Chromium instrument by 10x Genomics~\citep{zheng_massively_2017}. 

%% examples of cell types and lineages discovered


%% advantages and disadvantages of using scRNA-seq, and how it can be complemented
%Advantages: more molecules profiled, unbiased, 
%Disad: more expensive, batch effects, it's "just" RNA
%Mention: multi-omics (or other omics), spatial, sample multiplexing




%********************************** % Third Section  *************************************
\section{Methods for cell type classification}  %Section - 1.3 
\label{section1.3}

%%
While defining cell types through their transcriptome was not in the genesis of scRNA-seq, it was very early on envisioned. In 2011, Islam and colleagues end their discussion on their newly developed scRNA-seq method (STRT-seq) by stating "We envisage the future use of very large-scale single-cell transcriptional profiling to build a detailed map of naturally occurring cell types, which would give unprecedented access to the genetic machinery active in each type of cell at each stage of development."~\citep{islam_characterization_2011}. 

%%

%%



%********************************** % Fourth Section  *************************************
\section{Cell identity in the immune system}  %Section - 1.4 
\label{section1.4}

%% Introduction to the cells in the immune system


%% Introduction to helper T cells


%% regulation and plasticity of T cell identity




%********************************** % Fifth Section  *************************************
\section{Tissue-specific gene expression}  %Section - 1.5 
\label{section1.5}


%********************************** % Sixth Section  *************************************
\section{Insights and scope of this thesis}  %Section - 1.6
\label{section1.6}





% CHANGE below is still some example text that might later be useful
The most famous equation in the world: $E^2 = (m_0c^2)^2 + (pc)^2$, which is 
known as the \textbf{energy-mass-momentum} relation as an in-line equation.

A {\em \LaTeX{} class file}\index{\LaTeX{} class file@LaTeX class file} is a file, which holds style information for a particular \LaTeX{}.

\begin{align}
CIF: \hspace*{5mm}F_0^j(a) = \frac{1}{2\pi \iota} \oint_{\gamma} \frac{F_0^j(z)}{z - a} dz
\end{align}

\nomenclature[z-cif]{$CIF$}{Cauchy's Integral Formula}                                % first letter Z is for Acronyms 
\nomenclature[a-F]{$F$}{complex function}                                                   % first letter A is for Roman symbols
\nomenclature[g-p]{$\pi$}{ $\simeq 3.14\ldots$}                                             % first letter G is for Greek Symbols
\nomenclature[g-i]{$\iota$}{unit imaginary number $\sqrt{-1}$}                      % first letter G is for Greek Symbols
\nomenclature[g-g]{$\gamma$}{a simply closed curve on a complex plane}  % first letter G is for Greek Symbols
\nomenclature[x-i]{$\oint_\gamma$}{integration around a curve $\gamma$} % first letter X is for Other Symbols
\nomenclature[r-j]{$j$}{superscript index}                                                       % first letter R is for superscripts
\nomenclature[s-0]{$0$}{subscript index}                                                        % first letter S is for subscripts

\nomenclature[z-DEM]{DEM}{Discrete Element Method}
\nomenclature[z-FEM]{FEM}{Finite Element Method}
\nomenclature[z-PFEM]{PFEM}{Particle Finite Element Method}
\nomenclature[z-FVM]{FVM}{Finite Volume Method}
\nomenclature[z-BEM]{BEM}{Boundary Element Method}
\nomenclature[z-MPM]{MPM}{Material Point Method}
\nomenclature[z-LBM]{LBM}{Lattice Boltzmann Method}
\nomenclature[z-MRT]{MRT}{Multi-Relaxation 
Time}
\nomenclature[z-RVE]{RVE}{Representative Elemental Volume}
\nomenclature[z-GPU]{GPU}{Graphics Processing Unit}
\nomenclature[z-SH]{SH}{Savage Hutter}
\nomenclature[z-CFD]{CFD}{Computational Fluid Dynamics}
\nomenclature[z-LES]{LES}{Large Eddy Simulation}
\nomenclature[z-FLOP]{FLOP}{Floating Point Operations}
\nomenclature[z-ALU]{ALU}{Arithmetic Logic Unit}
\nomenclature[z-FPU]{FPU}{Floating Point Unit}
\nomenclature[z-SM]{SM}{Streaming Multiprocessors}
\nomenclature[z-PCI]{PCI}{Peripheral Component Interconnect}
\nomenclature[z-CK]{CK}{Carman - Kozeny}
\nomenclature[z-CD]{CD}{Contact Dynamics}
\nomenclature[z-DNS]{DNS}{Direct Numerical Simulation}
\nomenclature[z-EFG]{EFG}{Element-Free Galerkin}
\nomenclature[z-PIC]{PIC}{Particle-in-cell}
\nomenclature[z-USF]{USF}{Update Stress First}
\nomenclature[z-USL]{USL}{Update Stress Last}
\nomenclature[s-crit]{crit}{Critical state}
\nomenclature[z-DKT]{DKT}{Draft Kiss Tumble}
